plugin/src/main/java/net/dzikoysk/funnyguilds/rank/RankPlaceholdersService.java

```diff
@@ -16,49 +16,60 @@
 import net.dzikoysk.funnyguilds.user.UserRankManager;
 import net.dzikoysk.funnyguilds.user.top.UserTop;
 import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
 import panda.std.Option;
 
-public class RankUtils {
+public class RankPlaceholdersService {
 
     private static final Pattern TOP_PATTERN = Pattern.compile("\\{(PTOP|GTOP)-([A-Za-z_]+)-([0-9]+)}");
     private static final Pattern TOP_POSITION_PATTERN = Pattern.compile("\\{(POSITION|G-POSITION)-([A-Za-z_]+)}");
     private static final Pattern LEGACY_TOP_PATTERN = Pattern.compile("\\{(PTOP|GTOP)-([0-9]+)}");
 
+    private final FunnyGuilds plugin;
+
+    private final PluginConfiguration config;
+    private final MessageConfiguration messages;
+    private final TablistConfiguration tablistConfig;
+
+    private final UserRankManager userRankManager;
+    private final GuildRankManager guildRankManager;
+
+    public RankPlaceholdersService(FunnyGuilds plugin) {
```

Comment:
Tu przez konstruktor wszystko prosze, bez plugin. A jak potem potrzebujesz logger to albo dodaj go przez konstruktor albo zrób nowy dla tej klasy